src/server/authMiddleware.ts

```diff
@@ -45,8 +54,9 @@ const isPublicRoute = (req: NextRequest, options: MiddlewareOptions) => {
 		req.nextUrl.pathname
 	);
 	const isPublic = options.publicRoutes?.includes(req.nextUrl.pathname);
+	const isPrivate = options.privateRoutes?.includes(req.nextUrl.pathname);
 
-	return isDefaultPublicRoute || isPublic;
+	return isDefaultPublicRoute || isPublic || !isPrivate;
```

Comment:
the logic here is problematic

if user doesn't provide `privateRoutes` than `!isPrivate;` will be true, always

I think this is why test fails

we need to 
1. make sure that `privateRoutes` are defined
2. make sure that if someone provided both `publicRoutes` and `privateRoute` (they shouldn't, but they may do it, if they use non TS for example) we handle this gracefully - probably ignore `privateRoutes` in this case)

Reply:
I added logic so that the following happens - 

```
// If both publicRoutes and privateRoutes are provided, we prioritize publicRoutes
	if (options.publicRoutes && options.privateRoutes) {
		return isDefaultPublicRoute || isPublic;
	}

	// If only publicRoutes are provided
	if (options.publicRoutes) {
		return isDefaultPublicRoute || isPublic;
	}

	// If only privateRoutes are provided
	const isPrivate = options.privateRoutes?.includes(req.nextUrl.pathname);
	return isDefaultPublicRoute || !isPrivate;
```